File: scripts/upload-s3.js
#!/usr/bin/env node

const AWS = require('aws-sdk');
const fs = require('fs');
const path = require('path');

const bucketName = 'registro-participantes-dev-frontend-380012739300';
const distPath = path.join(__dirname, '../frontend/dist');

const s3 = new AWS.S3({
  region: 'us-east-1'
});

async function uploadDir(dirPath, s3Path = '') {
  const files = fs.readdirSync(dirPath);

  for (const file of files) {
    const filePath = path.join(dirPath, file);
    const s3Key = s3Path ? `${s3Path}/${file}` : file;

    if (fs.statSync(filePath).isDirectory()) {
      await uploadDir(filePath, s3Key);
    } else {
      const fileContent = fs.readFileSync(filePath);
      const params = {
        Bucket: bucketName,
        Key: s3Key,
        Body: fileContent,
        ContentType: getContentType(filePath)
      };

      try {
        await s3.upload(params).promise();
        console.log(`Uploaded: ${s3Key}`);
      } catch (err) {
        console.error(`Error uploading ${s3Key}:`, err);
      }
    }
  }
}

function getContentType(filePath) {
  const ext = path.extname(filePath).toLowerCase();
  const mimeTypes = {
    '.html': 'text/html',
    '.js': 'application/javascript',
    '.css': 'text/css',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpeg',
    '.svg': 'image/svg+xml',
    '.map': 'application/json'
  };
  return mimeTypes[ext] || 'application/octet-stream';
}

async function main() {
  console.log(`Uploading ${distPath} to s3://${bucketName}/...`);
  await uploadDir(distPath);
  console.log('Upload complete!');
}

main().catch(console.error);

File: scripts/upload_s3.py
#!/usr/bin/env python3
"""Script para subir frontend a S3"""

import boto3
import os
from pathlib import Path
import mimetypes
import sys

# Fix encoding en Windows
if sys.platform == 'win32':
    os.environ['PYTHONIOENCODING'] = 'utf-8'

bucket_name = 'registro-participantes-dev-frontend-380012739300'
dist_path = Path(__file__).parent.parent / 'frontend' / 'dist'
region = 'us-east-1'

s3_client = boto3.client('s3', region_name=region)

def get_content_type(file_path):
    """Obtener MIME type correcto para cada archivo"""
    ext = str(file_path).lower()

    # MIME types explícitos para archivos críticos
    mime_types = {
        '.html': 'text/html',
        '.js': 'application/javascript',
        '.mjs': 'application/javascript',
        '.css': 'text/css',
        '.json': 'application/json',
        '.png': 'image/png',
        '.jpg': 'image/jpeg',
        '.jpeg': 'image/jpeg',
        '.svg': 'image/svg+xml',
        '.gif': 'image/gif',
        '.ico': 'image/x-icon',
        '.woff': 'font/woff',
        '.woff2': 'font/woff2',
        '.ttf': 'font/ttf',
        '.eot': 'application/vnd.ms-fontobject',
        '.map': 'application/json'
    }

    # Buscar por extensión
    for ext_key, mime in mime_types.items():
        if ext.endswith(ext_key):
            return mime

    # Por defecto
    guessed, _ = mimetypes.guess_type(str(file_path))
    return guessed or 'application/octet-stream'

def upload_directory(local_path, s3_prefix=''):
    """Subir todos los archivos de un directorio a S3"""
    for root, dirs, files in os.walk(local_path):
        for file in files:
            file_path = Path(root) / file
            relative_path = file_path.relative_to(local_path)
            s3_key = f"{s3_prefix}/{relative_path}".lstrip('/').replace('\\', '/')

            # Determinar MIME type
            mime_type = get_content_type(file_path)

            # Subir archivo
            try:
                s3_client.upload_file(
                    str(file_path),
                    bucket_name,
                    s3_key,
                    ExtraArgs={'ContentType': mime_type}
                )
                print(f"[+] Uploaded: {s3_key} (type: {mime_type})")
            except Exception as e:
                print(f"[-] Error uploading {s3_key}: {e}")

def main():
    if not dist_path.exists():
        print(f"Error: {dist_path} no existe")
        return

    print(f"Subiendo {dist_path} a s3://{bucket_name}/...")
    upload_directory(dist_path)
    print("¡Upload completado!")
    print(f"\nAccede a: https://d3tfsu5pihpnue.cloudfront.net/")

if __name__ == '__main__':
    main()
